tests/lib/openstack_query/handlers/test_client_side_handler_datetime.py

```diff
@@ -33,147 +30,138 @@ def setUp(self):
-            },
-            "older_by_1_second": {
-                "prop": "2023-06-04T10:30:00Z",
-                "days": 0,
-                "hours": 0,
-                "minutes": 0,
-                "seconds": 1,
-            },
-            "equal": {
-                "prop": "2023-06-04T10:00:00Z",
-                "days": 0,
-                "hours": 0,
-                "minutes": 30,
-                "seconds": 0,
-            },
-            "younger_by_1_day": {
-                "prop": "2023-06-02T10:30:00Z",
-                "days": 1,
-                "hours": 0,
-                "minutes": 0,
-                "seconds": 0,
-            },
-            "younger_by_12_hours": {
-                "prop": "2023-06-03T10:30:00Z",
-                "days": 0,
-                "hours": 12,
-                "minutes": 0,
-                "seconds": 0,
-            },
-            "younger_by_1_second": {
-                "prop": "2023-06-03T10:29:59Z",
-                "days": 0,
-                "hours": 0,
-                "minutes": 0,
-                "seconds": 1,
-            },
-        }
-
-    @parameterized.expand(
-        [(f"test {preset.name}", preset) for preset in QueryPresetsDateTime]
-    )
-    def test_check_supported_all_presets(self, mock_current_time, _, preset):
+    def test_check_supported_all_presets(self, *_):
         """
         Tests that client_side_handler_datetime supports all datetime QueryPresets
         """
-        self.assertTrue(self.instance.check_supported(preset, MockProperties.PROP_1))
-
-    @parameterized.expand(
-        [
-            ("older_by_1_day", True),
-            ("older_by_12_hours", True),
-            ("older_by_1_second", True),
-            ("equal", False),
-            ("younger_by_1_day", False),
-            ("younger_by_12_hours", False),
-            ("younger_by_1_second", False),
-        ]
-    )
-    def test_prop_older_than(self, mock_current_time, name, expected_out):
+        self.assertTrue(
+            self.instance.check_supported(preset, MockProperties.PROP_1)
+            for preset in QueryPresetsDateTime
+        )
+
+    @staticmethod
+    def _run_prop_case_with_mocks(
+        method: Callable,
+        mock_datetime: Mock,
+        mock_time_get_timestamp: Mock,
+        prop_time: int,
+        user_time: int,
+    ):
+        """
+        Runs the test case with mocks to ensure the prop time handling works as expected
+        Does not assert the final result, that is for the test case to do
+        """
+        prop_str = NonCallableMock()
+        days, hours, mins, secs = (
+            NonCallableMock(),
+            NonCallableMock(),
+            NonCallableMock(),
+            NonCallableMock(),
+        )
+        mock_datetime.strptime.return_value.timestamp.return_value = (
+            prop_time  # Prop timestamp
+        )
+        mock_time_get_timestamp.return_value = (
+            user_time  # The user's selected timeframe
```

Comment:
Doesn't `prop_time` and `user_time` need to be `NonCallableMock()` as well?

Reply:
This is a small static method to remove the duplication in the tests. 

Since it runs for both the younger than or older than we need to pick real values for `<` or `>` , since `NonCallableMock > NonCallableMock` potentially doesn't make sense

I had to double check it was intended and not a bug / thing we could improve, but I can't think how to avoid it :thinking: